presto-tests/src/test/java/com/facebook/presto/server/TestCatalogResource.java

```diff
@@ -0,0 +1,212 @@
+            throws Exception
+    {
+        client = new JettyHttpClient();
+        runner = createQueryRunner();
+        server = runner.getCoordinator();
+    }
+
+    @AfterClass(alwaysRun = true)
+    public void teardown()
+    {
+        closeQuietly(server);
+        closeQuietly(client);
+        server = null;
+        client = null;
+    }
+
+    @Test
+    public void testCatalogResource()
+            throws Exception
+    {
+        final Map<String, String> catalogProperties = getCatalogPropertiesMap();
+        URI uri = uriBuilderFrom(server.getBaseUrl().resolve(format(CATALOG_RESOURCE_URI, "tpch_1"))).build();
+
+        // CREATE FLOW
+        Request.Builder requestBuiler = setContentTypeHeaders(false, preparePost());
+        Request request = createRequest(requestBuiler, catalogProperties, uri);
+        assertEquals(client.execute(request, createStringResponseHandler()).getStatusCode(), 201);
+        assertEquals(client.execute(request, createStringResponseHandler()).getStatusCode(), 409);
+        validateCatalogDetails(ImmutableSet.of("system", "testing_catalog", "tpch", "tpch_1"));
+
+        // DELETE FLOW
+        requestBuiler = setContentTypeHeaders(false, prepareDelete());
+        request = createRequest(requestBuiler, catalogProperties, uri);
+        assertEquals(client.execute(request, createStringResponseHandler()).getStatusCode(), 204);
+        assertEquals(client.execute(request, createStringResponseHandler()).getStatusCode(), 404);
+        validateCatalogDetails(ImmutableSet.of("system", "testing_catalog", "tpch"));
+
+        // UPDATE FLOW
+        requestBuiler = setContentTypeHeaders(false, preparePut());
+        request = createRequest(requestBuiler, catalogProperties, uri);
+        assertEquals(client.execute(request, createStringResponseHandler()).getStatusCode(), 201);
+        assertEquals(client.execute(request, createStringResponseHandler()).getStatusCode(), 204);
+        validateCatalogDetails(ImmutableSet.of("system", "testing_catalog", "tpch", "tpch_1"));
+
+        request = prepareGet().setUri(server.getBaseUrl().resolve("/v1/catalog")).build();
+        List<String> catalogList = client.execute(request, createJsonResponseHandler(LIST_CODEC));
+        assertTrue(!catalogList.isEmpty());
+        assertTrue(catalogList.contains("tpch_1"));
+    }
+
+    @Test
+    public void testInvalidCatalogName()
+    {
+        final Map<String, String> catalogProperties = getCatalogPropertiesMap();
+        URI uri = uriBuilderFrom(server.getBaseUrl().resolve(format(CATALOG_RESOURCE_URI, "UNKNOWN88"))).build();
+        Request.Builder requestBuiler = setContentTypeHeaders(false, preparePost());
+        Request request = createRequest(requestBuiler, catalogProperties, uri);
+        assertEquals(client.execute(request, createStringResponseHandler()).getStatusCode(), 404);
+    }
+
+    @Test
+    public void testConcurrentQueryRemoveCatalog() throws InterruptedException
```

Comment:
@infvg Check out https://github.com/aaneja/presto/commit/bfafae42097eb22b922814d776e57621efe4efc8. I fixed the code and test related to this

Reply:
You also need to set `@Test(singleThreaded = true)` on this class. Otherwise you'd have catalogs being added/removed in parallel by different tests